crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,132 @@
+                            "Worker Connection History",
+                            Style::default()
+                                .fg(Color::White)
+                                .add_modifier(Modifier::BOLD),
+                        ))
+                        .borders(Borders::ALL),
+                )
+                .x_axis(
+                    Axis::default()
+                        .title("time ->")
+                        .style(Style::default().fg(Color::Gray))
+                        .bounds([
+                            get_time_as_secs().unwrap_or(0) as f64
+                                - self.view_size.as_secs() as f64,
+                            get_time_as_secs().unwrap_or(0) as f64,
+                        ]),
+                )
+                .y_axis(
+                    Axis::default()
+                        .style(Style::default().fg(Color::Gray))
+                        .bounds([0.0, self.get_max_workers_in_window().unwrap_or(0) as f64])
+                        .labels(vec![
+                            Span::from(0.to_string()),
+                            Span::from(
+                                (self.get_max_workers_in_window().unwrap_or(0) / 2).to_string(),
+                            ),
+                            Span::from((self.get_max_workers_in_window().unwrap_or(0)).to_string()),
+                        ]),
+                );
+            frame.render_widget(chart, rect);
+        } else {
+            let header_text = vec![Spans::from("No workers connected")];
+            let text_block = Block::default()
+                .borders(Borders::ALL)
+                .style(Style::default().bg(Color::Black));
+            let paragraph = Paragraph::new(header_text)
+                .block(text_block)
+                .alignment(Alignment::Left)
+                .wrap(Wrap { trim: true });
+            frame.render_widget(paragraph, rect);
+        }
+    }
+
+    /// Returns the maximum number of workers connected in the past n seconds specified by view_size
+    fn get_max_workers_in_window(&self) -> Option<u32> {
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map(|current_time| {
+                self.worker_count_vec
+                    .iter()
+                    .filter(|count| {
+                        count.at_time.duration_since(UNIX_EPOCH).unwrap().as_secs()
+                            > current_time.as_secs() - self.view_size.as_secs()
+                    })
+                    .max_by(|a, b| (a.worker_count.cmp(&b.worker_count)))
+                    .map(|count| (count.worker_count))
+            })
+            .ok()
+            .unwrap_or(None)
+    }
+}
+
+impl Default for ClusterOverviewChart {
+    fn default() -> Self {
+        Self {
+            worker_count_vec: vec![],
+            view_size: Duration::new(30, 0),
+        }
+    }
+}
+
+fn get_time_as_secs() -> Option<u64> {
```

Comment:
Just `unwrap` this. It should be super rare and its inside the client, so it's not critical if it crahes.